### DISCUSSION TITLE
I get different results for resampling TCP versus menu.  -  Is this a bug?

### DISCUSSION THREAD

**Discussion by petersampsonaudacity** [audacity/audacity#4782] [Near future]
In the course of testing @Paul-Licameli 's   fixes for stereo tracks being made from mono tracks with mis-matched sample rates.

I have no idea if this is a bug or expected behavior - but it looks most odd to me.

Steps to reproduce:
1. Import a two minute stereo song with default sample rate 44100 Hz
2. Split stereo to mono
3. In the top track use the Track Control Panel to change the sample rate to 96000Hz
4. Select the bottom track
5. Tracks > Resample...
6. Select 96000 Hz
7. Observe: top track is reduced to just under half length - but bottom track remains at full 2 minute length

Now I was filly expecting both tracks to be the same half-length.  Is my expectation wrong,

![image](https://github.com/audacity/audacity/assets/50205984/a0bf7d35-24d5-4cca-b5ab-399211755e65)

I tested this on W10 with 3.3.3, latest 3.4.0 alpha and 2.4.2 and got the same results each time.

**Comment by Paul-Licameli** [comment]
For whatever reason, this has been the behavior for ages.  The pull-down menu does real resampling, but the context menu just reassigns the rate.

I don't know who first did it that way and why.

Making two context menu items might be what we need.

**Reply by LWinterberg** [reply]
> Making two context menu items might be what we need.

agreed, I've added that to the design for multi-track recording

![image](https://github.com/audacity/audacity/assets/87814144/6c605350-dee9-4222-8323-2ea4838113b3)

**Comment by DavidBailes** [comment]
> Now I was filly expecting both tracks to be the same half-length. Is my expectation wrong,

https://manual.audacityteam.org/man/audio_track_dropdown_menu.html#rate

The potential confusion was part of the motivation for this comment: https://github.com/audacity/audacity/pull/4739#pullrequestreview-1456850366

**Comment by petersampsonaudacity** [comment]
>Making two context menu items might be what we need.

The **_BIG_** downside of that is that you can only resample one track at a time.

Tracks > Resample instead allows the user to select multiple tracks or even the whole project for "resampling"

**Comment by petersampsonaudacity** [comment]
I note that a careful read of the manual shows:

a) using the TCP resampling changes the length of the audio
https://alphamanual.audacityteam.org/man/Audio_Track_Dropdown_Menu
> Increasing the rate squeezes the existing samples into a shorter length (so increasing playback speed and pitch), while decreasing the rate spreads them out over a greater length (so reducing playback speed and pitch).

b) using Tracks > Resample resamples **without** changing the length of the audio
https://alphamanual.audacityteam.org/man/Tracks_Menu
>Allows you to resample the selected track(s) to a new sample rate for use in the project, leaving the length (and hence playback speed and pitch unchanged.

**_BUT_** I still find these seemingly disparate behaviors most odd and unsettling

**Comment by petersampsonaudacity** [comment]
I have edited those two pages to try to clarify the differences between **rate change**  and **resampling**:

a) **change of rate** with the TCP dropdown menu
https://alphamanual.audacityteam.org/man/Audio_Track_Dropdown_Menu#Rate

b) **resampling** with Tracks > Resample...
https://alphamanual.audacityteam.org/man/Tracks_Menu#resample
